general/mem_manager.hpp

```diff
@@ -137,16 +141,18 @@ class Memory
       VALID_DEVICE  = 1 << 5, ///< %Device pointer is valid
       USE_DEVICE    = 1 << 6, /**< Internal device flag, see e.g.
                                    Vector::UseDevice() */
-      ALIAS         = 1 << 7  ///< Pointer is an alias
+      ALIAS         = 1 << 7, ///< Pointer is an alias
+      USE_TEMPORARY = 1 << 8  ///< Temporary Device memory flag
+
    };
 
    /// Pointer to host memory. Not owned.
    /** The type of the pointer is given by the field #h_mt; it can be any type
        from MemoryClass::HOST. */
-   T *h_ptr;
-   int capacity; ///< Size of the allocated memory
-   MemoryType h_mt; ///< Host memory type
-   mutable unsigned flags; ///< Bit flags defined from the #FlagMask enum
+   T *h_ptr{nullptr};
```

Comment:
MFEM's style is to initialize the variables in the constructor. We could change it, but not just for a few of them.

Reply:
Switched to `=` initializers !